mp4/samplegroupentries.go

```diff
@@ -149,3 +152,147 @@ func (s *UnknownSampleGroupEntry) Info(w io.Writer, specificBoxLevels, indent, i
 	}
 	return bd.err
 }
+
+// RollSampleGroupEntry - Grouping Type "roll"
+type RollSampleGroupEntry struct {
+	RollDistance int16
+}
+
+func DecodeRollSampleGroupEntry(name string, length uint32, sr *SliceReader) (SampleGroupEntry, error) {
+	entry := &RollSampleGroupEntry{}
+	entry.RollDistance = sr.ReadInt16()
+	return entry, nil
+}
+
+func (s *RollSampleGroupEntry) Type() string {
+	return "roll"
+}
+
+func (s *RollSampleGroupEntry) Size() uint64 {
+	return 2
+}
+
+func (s *RollSampleGroupEntry) Encode(sw *SliceWriter) {
+	sw.WriteInt16(s.RollDistance)
+}
+
+// Info - write box info to w
+func (s *RollSampleGroupEntry) Info(w io.Writer, specificBoxLevels, indent, indentStep string) (err error) {
+	bd := newInfoDumper(w, indent, s, -2)
+	bd.write(" * rollDistance: %d", s.RollDistance)
+	return bd.err
+}
+
+// RapSampleGroupEntry - Grouping Type "rap "
+type RapSampleGroupEntry struct {
```

Comment:
Spell out: "Random Access Point" Sample Group 
It is from ISO/IEC 14496-12 Ed. 6 2020 Section 10.4